src/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -751,9 +724,43 @@ private static unsafe int ScaleResult(Buf24* bufRes, uint iHiRes, int iScale)
                 throw new OverflowException(SR.Overflow_Decimal);
             }
 
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private static unsafe uint DivByConst(uint* rgulRes, uint iHiRes, out uint quotient, out uint remainder, uint power)
+            {
+                uint high = rgulRes[iHiRes];
+                remainder = high - (quotient = high / power) * power;
+                for (uint i = iHiRes - 1; (int)i >= 0; i--)
+                {
+#if BIT64
+                    ulong num = rgulRes[i] + ((ulong)remainder << 32);
+                    remainder = (uint)num - (rgulRes[i] = (uint)(num / power)) * power;
+#else
+                    // 32-bit RyuJIT doesn't convert 64-bit division by constant into multiplication by reciprocal. Do half-width divisions instead.
+                    Debug.Assert(power <= ushort.MaxValue);
+#if BIGENDIAN
```

Comment:
We prefer runtime check using `BitConverter.IsLittleEndian` to make the sources compatible with Mono. (Mono does not use ifdefs for BIGENDIAN.)

Reply:
`#if BIGENDIAN` is used in various places in this file and this is for CoreRT + CoreCLR only at first. Maybe compatibility with Mono can be a separate issue?

Reply:
Ok, I have not realized that BIGENDIAN is used in number of places.